packages/nx/src/native/tui/components/layout_manager.rs

```diff
@@ -0,0 +1,964 @@
+                Constraint::Length(terminal_pane_width),
+            ])
+            .split(area);
+
+        let task_list_area = chunks[0];
+        let terminal_pane_area = chunks[2]; // Skip the padding area (chunks[1])
+
+        let terminal_panes = match self.pane_arrangement {
+            PaneArrangement::None => Vec::new(),
+            PaneArrangement::Single => vec![terminal_pane_area],
+            PaneArrangement::Double => {
+                // For two panes, split the terminal pane area vertically
+                let pane_chunks = Layout::default()
+                    .direction(Direction::Vertical)
+                    .constraints([Constraint::Percentage(50), Constraint::Percentage(50)])
+                    .split(terminal_pane_area);
+                vec![pane_chunks[0], pane_chunks[1]]
+            }
+        };
+
+        LayoutAreas {
+            task_list: Some(task_list_area),
+            terminal_panes,
+        }
+    }
+
+    /// Creates a LayoutConfig from the current internal state.
+    pub fn create_config(&self) -> LayoutConfig {
+        LayoutConfig {
+            mode: self.mode,
+            task_list_visibility: self.task_list_visibility,
+            pane_arrangement: self.pane_arrangement,
+            task_count: self.task_count,
+        }
+    }
+
+    /// Sets the minimum width required for a horizontal layout.
+    pub fn set_min_horizontal_width(&mut self, width: u16) {
+        self.min_horizontal_width = width;
+    }
+
+    /// Sets the minimum height required for a vertical layout.
+    pub fn set_min_vertical_height(&mut self, height: u16) {
+        self.min_vertical_height = height;
+    }
+
+    /// Determines if a vertical layout is preferred based on terminal dimensions and tasks.
+    ///
+    /// This is used in Auto mode to decide between vertical and horizontal layouts.
+    ///
+    /// Factors that influence the decision:
+    /// - Terminal aspect ratio
+    /// - Number of tasks (single task prefers vertical layout)
+    /// - Minimum dimensions requirements
+    fn is_vertical_layout_preferred(
+        &self,
+        terminal_width: u16,
+        terminal_height: u16,
+        task_count: usize,
+    ) -> bool {
+        // If there's only a single task, prefer vertical layout
+        if task_count <= 1 {
+            return true;
+        }
+
+        // Calculate aspect ratio (width/height)
+        let aspect_ratio = terminal_width as f32 / terminal_height as f32;
+
+        // If very wide and not very tall, prefer horizontal
+        if aspect_ratio > 2.0 && terminal_height < self.min_vertical_height {
+            return false;
+        }
+
+        // If very tall and not very wide, prefer vertical
+        if aspect_ratio < 1.0 && terminal_width < self.min_horizontal_width {
+            return true;
+        }
+
+        // Otherwise, prefer horizontal for wider terminals, vertical for taller ones
+        aspect_ratio < 1.5
```

Comment:
I think this gets simpler if we look at it this way:

There are constraints which force vertical or horizontal if there's not enough space.

If the terminal area is within those constraints then an aspect ratio of 1.5 is basically where we draw the line between preferring vertical vs horizontal.